src/parser.ml

```diff
@@ -303,15 +303,17 @@ let parser arg_ident =
 
 (** Metavariable identifier (regular or escaped, prefixed with ['?']). *)
 let parser meta =
-  | "?" - id:{regular_ident | escaped_ident} -> in_pos _loc id
+  | "?" - id:{regular_ident | escaped_ident} ->
+      if id = "_" then Earley.give_up (); in_pos _loc id
 
 (** Pattern variable identifier (regular or escaped, prefixed with ['&']). *)
 let parser patt =
-  | "&" - id:{regular_ident | escaped_ident} -> in_pos _loc id
+  | "&" - id:{regular_ident | escaped_ident} ->
+      if id = "_" then None else Some(in_pos _loc id)
```

Comment:
Perhaps we should exclude every ident starting with _.

Reply:
Again I'm not sure, there is no harm in having them. Anyway, we can always revise these syntax details later (they have little to do with this PR).

Reply:
But isn't it the case that, currently in your PR, if I write `\ _x,t` then I cannot use `_x` in `t`, and if I write `&_x` in a LHS then I can use `_x` in the RHS? This is not consistent.

Reply:
Currently, `_x` and `&_x` can be used. And there is no warning silencing for an unnecessary `&_x`.

Reply:
Why not changing this behavior?